src/main/java/org/jenkinsci/plugins/workflow/support/steps/input/InputStepExecution.java

```diff
@@ -157,7 +159,19 @@ public HttpResponse doSubmit(StaplerRequest request) throws IOException, Servlet
     @RequirePOST
     public HttpResponse doProceed(StaplerRequest request) throws IOException, ServletException, InterruptedException {
         preSubmissionCheck();
-        Map<String,Object> v = parseValue(request);
+        Object params = request.getSubmittedForm().get("parameter");
+        JSONArray v = params != null ? JSONArray.fromObject(params) : null;
+        return proceed(v, request);
+    }
+
+    /**
+     * Processes the acceptance (approval) request.
+     * @param params parameters element value, is array of map that represents the parameters sent in the request
+     * @param request stapler request
+     * @return A HttpResponse object that represents Status code (200) indicating the request succeeded normally.
+     */
+    public HttpResponse proceed(@Nullable JSONArray params, @Nonnull StaplerRequest request) throws ServletException, InterruptedException, IOException {
```

Comment:
@jglick JSONArray is List<Object>, if we use List<JSONObject> then caller has to do this conversion, makes it bit tedious. We could use List<Object> as well, but then it's better to be strongly typed and use JSONArray so that consumers know then need to pass specific JSON java type.